src/esp/sensor/CameraSensor.h

```diff
@@ -109,67 +117,72 @@ class CameraSensor : public VisualSensor {
     recomputeBaseProjectionMatrix();
   }  // CameraSensor::setFOV
 
-  /**
-   * @brief Returns the FOV of this CameraSensor
-   */
-  Mn::Deg getFOV() const {
-    float fov = std::atof(spec_->parameters.at("hfov").c_str());
-    return Mn::Deg{fov};
-  }
-
   /**
    * @brief Sets camera type and calculates appropriate size vector for
    * display.
    */
   void setCameraType(const SensorSubType& _cameraType) {
-    spec_->sensorSubType = _cameraType;
+    CORRADE_ASSERT(_cameraType == SensorSubType::Pinhole ||
+                       _cameraType == SensorSubType::Orthographic,
+                   "CameraSensor::setCameraType(): _cameraType is not "
+                   "SensorSubType "
+                   "Pinhole or Orthographic", );
+    cameraSensorSpec_->sensorSubType = _cameraType;
     recomputeBaseProjectionMatrix();
   }  // CameraSensor::setCameraType
 
   /**
    * @brief Returns the camera type of this Sensor
    */
-  SensorSubType getCameraType() const { return spec_->sensorSubType; }
+  SensorSubType getCameraType() const {
+    return cameraSensorSpec_->sensorSubType;
+  }
 
   /**
    * @brief Sets width of this sensor's view port
    */
-  void setWidth(int _width) {
-    spec_->resolution[1] = _width;
-    width_ = _width;
+  void setWidth(int width) {
+    CORRADE_ASSERT(width > 0,
+                   "CameraSensor::setWidth(): resolution height and "
+                   "width must be greater than 0", );
+    cameraSensorSpec_->resolution[1] = width;
     recomputeBaseProjectionMatrix();
   }
-  int getWidth() const { return width_; }
 
   /**
    * @brief Sets height of this sensor's view port
    */
-  void setHeight(int _height) {
-    spec_->resolution[0] = _height;
-    height_ = _height;
+  void setHeight(int height) {
+    CORRADE_ASSERT(height > 0,
+                   "CameraSensor::setHeight(): resolution height and "
+                   "width must be greater than 0", );
+    cameraSensorSpec_->resolution[0] = height;
     recomputeBaseProjectionMatrix();
   }
-  int getHeight() const { return height_; }
 
   /**
    * @brief Sets near plane distance.
    */
   void setNear(float _near) {
-    spec_->parameters.at("near") = std::to_string(_near);
+    CORRADE_ASSERT(_near > 0,
+                   "CameraSensor::setNear(): near plane distance must be "
+                   "greater than 0", );
     near_ = _near;
     recomputeBaseProjectionMatrix();
   }
-  float getNear() { return near_; }
+  float getNear() const { return near_; }
 
   /**
    * @brief Sets far plane distance.
    */
   void setFar(float _far) {
-    spec_->parameters.at("far") = std::to_string(_far);
+    CORRADE_ASSERT(
```

Comment:
@vauduong :
This is something we did not capture.
"near" and "far" should be in the `VisualSensor` spec! They were in the spec before refactoring.
 Then member variables, `far_` and `near_` should be removed.
 Please do fix it in another PR.

Reply:
@vauduong : Also, please lift up the 
  `bool getObservation(sim::Simulator& sim, Observation& obs) override;`
  `bool getObservationSpace(ObservationSpace& space) override;`
  from `CameraSensor` to `VisualSensor`

Reply:
move `ortho_scale` to the `CameraSensorSpec`.
move `channel` and `observationSpace` to `VisualSensorSpec`. (@Skylion007 : is `observationSpace` used somewhere in the python side??)

Reply:
It's replicated entirely on the Python side. The C++ isn't used directly.

Reply:
deprecate the `encoding` (I did not see it was used anywhere, please double check).
deprecate the `observationSpace`.

Reply:
@vauduong : you can see the branch "cubemap". See what I did there for your reference.